pymatgen/command_line/bader_caller.py

```diff
@@ -193,3 +206,175 @@ def get_oxidation_state_decorated_structure(self):
                    for i in range(len(structure))]
         structure.add_oxidation_state_by_site(charges)
         return structure
+
+    @property
+    def result(self):
+
+        summary = {
+            "min_dist": [d['min_dist'] for d in self.data],
+            "charge": [d['charge'] for d in self.data],
+            "volume": [d['atomic_vol'] for d in self.data],
+            "vacuum_charge": self.vacuum_charge,
+            "vacuum_volume": self.vacuum_volume,
+            "reference_used": self.reference_used,
+            "bader_version": self.version,
+        }
+
+        if self.potcar:
+            charge_transfer = [self.get_charge_transfer(i) for i in range(len(self.data))]
+            summary['charge_transfer'] = charge_transfer
+
+        return BaderResult(summary)
+
+
+class BaderResult(MSONable):
```

Comment:
I am not convinced of the value of this. A dict is already jsonable without modifications. Why do we need a separate class to store the results? I can see the value if you actually want to combine results in some useful way, e.g., incorporating it with the relaxed structure to provide a propertied structure, or provide more sophisticated analyses.

The from_path and from_object methods can easily be standalone functions. 

The methods below also assumes certain names for CHGCAR, AECAR0 etc. It would be better to glob files of CHGCAR* and choose the sorted version. Often, the files we use have suffixes like CHGCAR.relax.2.gz.